src/main/java/org/broadinstitute/hellbender/tools/spark/sv/AssemblyAlignmentParser.java

```diff
@@ -145,8 +149,14 @@
                     final Cigar memoryCigar = new Cigar(cigarMemoryList);
                     final int effectiveReadLen = memoryCigar.getReadLength() + SVVariantCallerUtils.getTotalHardClipping(memoryCigar) - SVVariantCallerUtils.getNumClippedBases(true, memoryCigar);
 
-                    // infer reference interval
-                    final SimpleInterval referenceInterval = new SimpleInterval(oneRegion.referenceInterval.getContig(), refIntervalStart, refIntervalStart + memoryCigar.getReferenceLength()-1);
+                    // infer reference interval taking into account of strand
+                    final int refBoundary2ndInTheDirectionOfContig = refBoundary1stInTheDirectionOfContig + (oneRegion.forwardStrand ? memoryCigar.getReferenceLength()-1 : -memoryCigar.getReferenceLength()+1) ;
```

Comment:
Can you make the last part of the ternary be explicitly `-1 * memoryCigar.getReferenceLength() + 1`?